starboard/shared/starboard/link_receiver.cc

```diff
@@ -35,81 +37,113 @@ namespace shared {
 namespace starboard {
 
 namespace {
+
+#if defined(SOMAXCONN)
+const int kMaxConn = SOMAXCONN;
+#else
+const int kMaxConn = 128;
+#endif
+
 // Creates a socket that is appropriate for binding and listening, but is not
 // bound and hasn't started listening yet.
-std::unique_ptr<Socket> CreateServerSocket(SbSocketAddressType address_type) {
-  std::unique_ptr<Socket> socket(new Socket(address_type));
```

Comment:
This calls `SbSocketCreate`, and looking at `socket_create.cc` it also makes the socket nonblocking. I don't see that preserved in your change.